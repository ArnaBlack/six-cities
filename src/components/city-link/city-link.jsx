import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import ActionCreators from '../../store/actions/action-creators';

class CityLink extends PureComponent {
  constructor(props) {
    super(props);

    this._handleClick = this._handleClick.bind(this);
  }

  render() {
    const {
      city,
      isActive,
    } = this.props;

    const activeClass = isActive ? `tabs__item--active` : ``;

    return <a
      className={`locations__item-link tabs__item ${activeClass}`}
      href="#"
      onClick={this._handleClick}
    >
      <span>{city}</span>
    </a>;
  }

  _handleClick(evt) {
    evt.preventDefault();
    const {
      city,
      onClick,
    } = this.props;
    onClick(city);
  }
}

CityLink.propTypes = {
  city: PropTypes.string.isRequired,
  isActive: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onClick: (city) => dispatch(ActionCreators.changeCity(city)),
});

export {CityLink};
export default connect(null, mapDispatchToProps)(CityLink);
