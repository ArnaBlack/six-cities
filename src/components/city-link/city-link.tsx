import * as React from 'react';
import {connect} from 'react-redux';

import CityActionCreator from '../../store/city/action-creator/action-creator';

import {City} from '../../types';

interface Props {
  city: City,
  isActive: boolean,
  onCityClick: () => void,
  onClick: (city: City) => void,
}

class CityLink extends React.Component<Props, null> {
  constructor(props) {
    super(props);

    this._handleClick = this._handleClick.bind(this);
  }

  render() {
    const {
      city,
      isActive,
    } = this.props;
    const activeClass = isActive ? `tabs__item--active` : ``;

    return <a
      className={`locations__item-link tabs__item ${activeClass}`}
      href="#"
      onClick={this._handleClick}
    >
      <span>
        {city.name}
      </span>
    </a>;
  }

  _handleClick(evt) {
    evt.preventDefault();
    const {
      city,
      onCityClick,
      onClick,
    } = this.props;

    onCityClick();
    onClick(city);
  }
}

const mapDispatchToProps = (dispatch) => ({
  onClick: (city) => dispatch(CityActionCreator.changeCity(city)),
});

export {CityLink};
export default connect(null, mapDispatchToProps)(CityLink);
