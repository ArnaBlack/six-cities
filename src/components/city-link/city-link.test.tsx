import * as React from 'react';
import {
  shallow,
  configure,
} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {CityLink} from './city-link';

configure({adapter: new Adapter()});

const mock = {
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13,
    },
  },
  isActive: true,
};

it(`CityLink renders correctly`, () => {
  const {
    city,
    isActive
  } = mock;
  const onClick = jest.fn();
  const onCityClick = jest.fn();
  const tree = shallow(<CityLink
    city={city}
    isActive={isActive}
    onClick={onClick}
    onCityClick={onCityClick}
  />);

  expect(tree).toMatchSnapshot();
});
