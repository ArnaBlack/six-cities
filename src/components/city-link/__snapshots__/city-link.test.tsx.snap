// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityLink renders correctly 1`] = `ShallowWrapper {}`;
